apps/server/src/modules/learnroom/service/course.service.ts

```diff
@@ -9,4 +9,22 @@ export class CourseService {
 	async findById(courseId: EntityId): Promise<Course> {
 		return this.repo.findById(courseId);
 	}
+
+	public async deleteUserDataFromCourse(userId: EntityId): Promise<number> {
+		const [courses, count] = await this.repo.findAllByUserId(userId);
+
+		const updatedCourses = courses.map(
+			(course: Course) =>
+				({
+					...course,
+					students: course.students.remove((u) => u.id === userId),
+					teachers: course.teachers.remove((u) => u.id === userId),
+					substitutionTeachers: course.substitutionTeachers.remove((u) => u.id === userId),
```

Comment:
Could you please add methods for these operations in the `Course` class and move this logic there?

Reply:
done